scipy/sparse/_sputils.py

```diff
@@ -126,11 +126,12 @@ def getdtype(dtype, a=None, default=None):
                 raise TypeError("could not interpret data type") from e
     else:
         newdtype = np.dtype(dtype)
-        if newdtype == np.object_:
-            raise ValueError(
-                "object dtype is not supported by sparse matrices"
-            )
 
+    if newdtype not in supported_dtypes:
+        supported_dtypes_fmt = ", ".join([t.__name__ for t in supported_dtypes])
+        raise ValueError(f"scipy.sparse does not support dtype {newdtype.name}. "
```

Comment:
Here we use `newdtype.name`, but above we use `t.__name__`. Is there a distinction between these two?

Reply:
There is, because the objects are comparable, but not the same - np.dtype vs numpy specific type class.
- `t` is an element of list of exact numpy type objects, which are derived from `generic` interface and have  `__name__` property.
- `np.dtype` is a little different, derived using Generics and TypeVar, and only has `name` property.

Reply:
Gotcha, thanks for the explanation!